### ISSUE TITLE
Split openpype lib into 2 or more parts

### ISSUE THREAD

**Issue by iLLiCiTiT** [pypeclub/OpenPype#1574]
## Description
We have to split content of openpype lib into core lib and all other lib parts. Only core lib should be imported on start before any logic in OpenPype is used to be able do validations and have ability in real library parts to use all imports without wrappers.

**Comment by antirotor** [comment]
also thing like otio (editorial.py) should be imported elsewhere, not in core or lib I think. That would solve some headaches from wrong builds of otio wheels and so on.

**Comment by iLLiCiTiT** [comment]
> also thing like otio (editorial.py) should be imported elsewher

Agree, core should have real minimum that would not call or use on import. That would give us ability to run validations on start and catch them properly.

**Comment by iLLiCiTiT** [comment]
So I moved files here and there trying to find out what should be and what should not be in core. Will try to sum up what I think about it.

 would split current lib into these parts
- `openpype core` - with these parts you can use OpenPype as software
    - mongo connection
    - logging definition
    - settings
- `pipeline core` - without these pipeline logic of OpenPype won't work
    - anatomy
    - applications
    - modifications of avalon and pyblish (currently in `plugin_tools.py`)
    - ...
- `openpype core utils` - handy functions to get information from or about core
    - pype info
    - execute commands
    - vendored binaries
    - ...
- `pipeline utils` - handy functions that can be used across multiple places in pipeline logic
    - profiles
    - avalon context functions (make sence to use it only in host where context is known and set in `avalon.Session`
    - ...
- `specific pipeline utils` - functions that care about one specific pipeline part
    - editorial
    - abstract implementations

**Comment by jezscha** [comment]
Regarding the editorial and its otio import: I believe the otio import could be hidden in decoration function and passed into a function as function argument. So during importing of lib from anywhere otio will not be imported until the time you will need to use any of editorial.py functiuons. Just an idea..  

```
def import_otio(func):
    def inner():
		import opentimelineio as otio
        func(otio=otio)
    return inner

@import_otio
def any_function(input_object, otio=None):
	if otio:
		otio_object = otio.create_object(input_object)
    	return otio_object
```

**Comment by iLLiCiTiT** [comment]
> Regarding the editorial and its otio import: I believe the otio import could be hidden in decoration function and passed into a function as function argument.

That is actually different issue (and there is a better solution). The problem is that in openpype.lib is used a lot of crutial parts from inside of openpype.lib which should be validated before usage. For example `PypeLogger` is created on many places on import which may cause crash before it is possible to validate if mongo logging can and should be used. Same for avalon repository and many others...
